### PR TITLE
Add a CD workflow

### PR DESCRIPTION
**Pull request by pedroCollogno**
## Feature
Add a CD workflow that will automatically upload new package releases to PyPi

## To Dos

- [x]  Rename existing build workflow
- [x] Add a release stage to PyPi
- [x] Add a CI test script that triggers on open PRs
- [x] Add the obvious safeguard `on: push: branches: - master` to only trigger deploys on merged PRs
- [x] Make this step only trigger on Releases/GH tags -> uncomment the "if" statement
- [ ] Automate version fetching/bumping (can be done with Poetry, [example here](https://forcepush.tech/python-package-ci-cd-with-git-hub-actions)) --> added to issue

### CODE REVIEW

File: .github/workflows/test-build.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
-name: Build distro
+name: Test built package
 
-on: push
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (axelpey) [inline .github/workflows/test-build.yml:6]: Du coup que je capte, cette action on l'utilise plus?

File: .github/workflows/build-release.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
+    name: >-
+      Publish Python 🐍 distribution 📦 to PyPI
+    # if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
```
Comment (axelpey) [inline .github/workflows/build-release.yml:36]: Okay go switcher sur le if asap :)

**Review by axelpey — APPROVED** [review]

File: .github/workflows/test-build.yml (hunk 1 above)
Comment (pedroCollogno) [reply .github/workflows/test-build.yml:6]: Juste quand la PR est pushed sur main

File: .github/workflows/build-release.yml (hunk 2 above)
Comment (pedroCollogno) [reply .github/workflows/build-release.yml]: done

**Comment by axelpey** [comment]
Lesgo

**Pull request merged**